components/proxy/src/test/kotlin/com/hotels/styx/services/YamlFileConfigurationServiceTest.kt

```diff
@@ -0,0 +1,634 @@
+import com.hotels.styx.infrastructure.configuration.json.ObjectMappers
+import com.hotels.styx.infrastructure.configuration.yaml.JsonNodeConfig
+import com.hotels.styx.routing.RoutingObjectFactoryContext
+import com.hotels.styx.routing.RoutingObjectRecord
+import com.hotels.styx.routing.db.StyxObjectStore
+import com.hotels.styx.routing.handlers.PathPrefixRouter
+import com.hotels.styx.routing.handlers.ProviderObjectRecord
+import com.hotels.styx.services.OriginsConfigConverter.Companion.OBJECT_CREATOR_TAG
+import io.kotlintest.Spec
+import io.kotlintest.eventually
+import io.kotlintest.matchers.collections.shouldContainAll
+import io.kotlintest.seconds
+import io.kotlintest.shouldBe
+import io.kotlintest.specs.FunSpec
+import kotlinx.coroutines.delay
+import org.slf4j.LoggerFactory
+import java.io.File
+import java.util.Optional
+
+class YamlFileConfigurationServiceTest : FunSpec() {
+    val tempDir = createTempDir(suffix = "-${this.javaClass.simpleName}")
+    val originsConfig = File("${tempDir.absolutePath}/config.yml")
+    val LOGGER = LoggerFactory.getLogger(YamlFileConfigurationServiceTest::class.java)
+
+    override fun beforeSpec(spec: Spec) {
+        LOGGER.info("Temp directory: " + tempDir.absolutePath)
+        LOGGER.info("Origins file: " + originsConfig.absolutePath)
+    }
+
+    override fun afterSpec(spec: Spec) {
+        tempDir.deleteRecursively()
+    }
+
+    private fun with(service: YamlFileConfigurationService, action: (YamlFileConfigurationService) -> Unit) {
+        try {
+            action(service)
+        } finally {
+            service.stop().join()
+        }
+    }
+
+    init {
+        context("When the service starts") {
+            test("It populates object store from the YAML configuration") {
+                val routeDb = StyxObjectStore<RoutingObjectRecord>()
+                val serviceDb = StyxObjectStore<ProviderObjectRecord>()
+
+                writeOrigins("""
+                    ---
+                    - id: "app"
+                      path: "/"
+                      origins:
+                      - { id: "app-01", host: "localhost:9090" }
+                      - { id: "app-02", host: "localhost:9091" }
+                    """.trimIndent())
+
+                with(YamlFileConfigurationService(
+                        routeDb,
+                        OriginsConfigConverter(serviceDb, RoutingObjectFactoryContext(objectStore = routeDb).get(), "origins-cookie"),
+                        YamlFileConfigurationServiceConfig(originsConfig.absolutePath),
+                        serviceDb)) {
+                    it.start().join()
+
+                    eventually(2.seconds, AssertionError::class.java) {
+                        routeDb.entrySet().size shouldBe 4
+                    }
+                }
+            }
+
+            // TODO: Can't be tested as FileChangeMonitor fails to start if the origins file doesn't exist.
```

Comment:
Do you mean to remove this test and consider this as a feature? I'm okay with that. However that behaviour feels counter intuitive with "monitor: true" flag.

Reply:
Yes... In other case I would suggest having that behaviour controlled by some configuration flag. My concern is that in cases with an incorrect origins file Styx might start correctly but never serve any requests... Maybe we can support that behaviour but it would require some changes elsewhere

Reply:
Okay. I will remove this test.